Simbody/include/simbody/internal/CableSpan.h

```diff
@@ -0,0 +1,358 @@
+    /** Set the maximum number of solver iterations allowed when enforcing the
+     * surface constraints of all obstacles. */
+    void setSurfaceProjectionMaxIter(int maxIter);
+
+    /** Get the accuracy used by the numerical integrator when computing a
+     * geodesic over an obstacle. */
+    Real getIntegratorAccuracy() const;
+    /** Set the accuracy used by the numerical integrator when computing a
+     * geodesic over an obstacle. */
+    void setIntegratorAccuracy(Real accuracy);
+
+    /** Get the maximum number of solver iterations for finding the optimal
+     * path. */
+    int getSolverMaxIter() const;
+    /** Set the maximum number of solver iterations for finding the optimal
+     * path. */
+    void setSolverMaxIter(int maxIter);
+
+    // TODO merge with surface constraint tolerance above?
+    Real getPathErrorAccuracy() const;
+    void setPathErrorAccuracy(Real accuracy);
+
+    // TODO advanced parameter: Max allowed step during Newton iteration in
+    // angles. Using the local curvatures the max angular step is converted to a
+    // max allowed translation.
+    Real getMaxRadialStepInDegrees() const;
+    void setMaxRadialStepInDegrees(Real maxStepInDeg);
+
+//------------------------------------------------------------------------------
+//                     CABLE CALCULATIONS
+//------------------------------------------------------------------------------
+
+    /** Get the total cable length.
+     * State must be realized to Stage::Position. */
+    Real getLength(const State& state) const;
+
+    /** Get the derivative of the total cable length.
+     * State must be realized to Stage::Position. */
+    Real getLengthDot(const State& state) const;
+
+    void applyBodyForces(
+        const State& state,
+        Real tension,
+        Vector_<SpatialVec>& bodyForcesInG) const;
+
+    /** Compute the cable power.
+     * State must be realized to Stage::Position. */
+    Real calcCablePower(const State& state, Real tension) const;
+
+    /** Compute points on the path spanned by this cable.
+     * The curve segments will be resampled from the underlying geodesic such
+     * that the computed points are at equal length intervals. The straight
+     * line segments are not resampled.
+     * State must be realized to Stage::Position.
+     *
+     * @param state System State.
+     * @param lengthInterval Each curve segments will be resampled such the the
+     * points interspacing does not exceed this value.
+     * @param sink Where the path points will be written to.
+     * @return The total number of computed points.
+     */
+    int calcPathPoints(
+        const State& state,
+        Real lengthInterval,
+        std::function<void(Real length, Vec3 point)> sink) const;
+
+    class Impl;
+
+private:
+    /** Wrap the Impl, increasing the reference count. */
+    explicit CableSpan(std::shared_ptr<Impl> impl): m_Impl(std::move(impl)) {}
```

Comment:
Seems unnecessary: each public constructor can do this when initializing the `m_Impl` member variable, rather than forwarding to a separate private ctor.